### PR TITLE
Fix infinite paged iterable bug.

### PR DESCRIPTION
**Pull request by johnou**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -20,12 +20,12 @@
   <parent>
     <groupId>com.cloudbees</groupId>
     <artifactId>cloudbees-oss-parent</artifactId>
-    <version>5</version>
+    <version>6</version>
```
Comment (johnou) [inline pom.xml]: You can ignore this, should be the same values but I had to merge master to my branch.

File: src/main/java/org/zendesk/client/v2/Zendesk.java [hunk 2]
```diff
@@ -1103,42 +1102,34 @@ public void updateInstallation(int id, String json) {
...
-    private Request req(String method, Uri template, int page) {
+    private Request req(String method, Uri template, String contentType, byte[] body) {
         RequestBuilder builder = new RequestBuilder(method);
         if (realm != null) {
             builder.setRealm(realm);
         } else {
             builder.addHeader("Authorization", "Bearer " + oauthToken);
         }
-        builder.addQueryParameter("page", Integer.toString(page));
-        builder.setUrl(template.toString().replace("%2B", "+")); //replace out %2B with + due to API restriction
```
Comment (johnou) [inline src/main/java/org/zendesk/client/v2/Zendesk.java]: optimized API restriction replacement as well, now it doesn't compile a new pattern every request.

File: src/main/java/org/zendesk/client/v2/Zendesk.java [hunk 3]
```diff
@@ -1431,48 +1416,34 @@ public static ObjectMapper createMapper() {
...
             this.url = url;
-            this.initialPage = initialPage;
         }
 
         public Iterator<T> iterator() {
-            return new PagedIterator(initialPage);
+            return new PagedIterator(url);
         }
 
         private class PagedIterator implements Iterator<T> {
```
Comment (johnou) [inline src/main/java/org/zendesk/client/v2/Zendesk.java]: Unfortunately PagedIterator is too tightly coupled to Zendesk and I didn't have time to refactor / extract it.

Comment (stephenc) [inline src/main/java/org/zendesk/client/v2/Zendesk.java]: I don't see that it should be decoupled from Zendesk necessarily

**Pull request merged**